pkg/provider/provider.go

```diff
@@ -458,20 +473,32 @@ func (k *kubeProvider) Update(
 	newInputs := propMapToUnstructured(newResInputs)
 
 	// Apply update.
-	liveObj, err := await.Update(k.pool, k.client, oldInputs, newInputs)
-	if err != nil {
-		return nil, err
+	initialized, awaitErr := await.Update(k.pool, k.client, oldInputs, newInputs)
+	if awaitErr != nil {
+		var getErr error
+		initialized, getErr = k.readLiveObject(newInputs)
+		if getErr != nil {
+			// Object update/creation failed.
+			return nil, awaitErr
+		}
+		// If we get here, resource successfully created, but failed to initialize.
```

Comment:
`created => `updated`?

More generally, "created" here and above means something like "was received by the Kubernetes master" and "failed to initialize" means something like "failed to reach the expected healthy state for the resource" - right?